src/bin/js/worker_js.ml

```diff
@@ -120,6 +120,22 @@ let main worker_id content =
       Frontend.print_status status n
     in
 
+    let print_model ppf env =
+      let ur = SAT.get_unknown_reason env in
+      Printer.print_fmt (Options.Output.get_fmt_diagnostic ())
+        "@[<v 0>; Returned unknown reason = %a@]"
+        Sat_solver_sig.pp_unknown_reason_opt ur;
+      match SAT.get_model env with
+      | None ->
+        Printer.print_fmt (Options.Output.get_fmt_diagnostic ())
+          "@[<v 0>It seems that no model has been computed so \
+           far. You may need to change your model generation strategy \
+           or to increase your timeouts."
+
+      | Some (lazy model) ->
+        Models.output_concrete_model ppf model
+    in
+
```

Comment:
All the remarks made in the previous round on the version of this function in `solving_loop.ml` apply. Please either make the same changes as in `solving_loop.ml` or (better, because it would avoid this type of issues) keep the implementation in the library and call it from both `solving_loop.ml` and `worker.ml`.

Reply:
I know ;) actually it's a bit annoying to factorize this function because of the presence of the first-class module as argument. I've several options but none of them are really satisfying. If I move back the function `print_model` in the module `Frontend`, I will need to recreate the module `Frontend` in the `get-model` handler as this function requires the module `SAT`. 
```ocaml 
          let () = match State.get partial_model_key st with
            | Some (Model ((module SAT), env)) ->
              let module FE = Frontend.Make (SAT) in
              FE.print_model (Options.Output.get_fmt_regular ()) env
            | None ->
              (* TODO: add the location of the statement. *)
              recoverable_error "No model produced."
          in
          st
```
I don't know what's the overhead of recreate a module here.

I can also retrieve the model in the `Solving_loop` and feed the function `print_model` with its return but `print_model` is just a wrapper around the pretty print of models to display a warning if we don't produce a model as we should according to the option `produce-models` and the status of the last `check-sat`. Thus, we have to feed the function `print_model` with an `option`, which is a bit ugly.

Reply:
I factorize the code using the first method I described.

Reply:
There is no references in the frontend so it would be fine to re-create a `Frontend.Make` here, and the cost would be virtually zero compared to actually computing the model.

Having a `print_model_opt` in `Frontend` (not `Frontend.Make`) would also work. `Option.map` is not ugly :)

Reply:
Actually `print_model` in `Frontend` could take the first-class module, which would probably be the best solution.

Reply:
I thought I could do that but `Frontend` is already a functor taking the `SAT` module as argument. Besides, if I want to do that, I will need to move the gadt you created in `Solving_loop` into `Frontend` and expose it here. I prefer using the function `Frontend.Make`. If the cost is negligible, it's acceptable.